master/buildbot/changes/gitpoller.py

```diff
@@ -311,6 +312,16 @@ def _convert_nonzero_to_failure(res,
             "utility to handle the result of getProcessOutputAndValue"
             (stdout, stderr, code) = res
             if code != 0:
+                if code == 128:
+                    # 128 is the error code git returns when things are
+                    # wrong on its end
+                    error_message = ('RuntimeError: command %s %s in %s on '
+                                    'repourl %s failed with exit code %d: %s'
+                                    % (command, args, path, self.repourl,
+                                       code, stderr)).strip()
+                    sys.stderr.write(error_message)
+                    sys.stderr.flush()
+                    sys.exit(1)
```

Comment:
It should, shouldn't it? We are just trying to replicate an exception without actually providing a traceback. Before the PR, the exception was already halting the build since the poller couldn't fetch the change. We're doing the same here, minus the long traceback.